app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js

```diff
@@ -27,7 +27,7 @@ const switchEthereumChain = {
 export default switchEthereumChain;
 
 function findExistingNetwork(chainId, findNetworkConfigurationBy) {
-  if (chainId in CHAIN_ID_TO_TYPE_MAP) {
+  if (chainId in CHAIN_ID_TO_TYPE_MAP && chainId !== CHAIN_IDS.LINEA_TESTNET) {
```

Comment:
This condition is meant to select for the built-in Infura Mainnet network, and any other built-in Ethereum testnets. It would be better to update this condition to better reflect that requirement, rather than adding an exception for Linea.

We have shared constants for these networks already: `INFURA_PROVIDER_TYPES` and `BUILT_IN_INFURA_NETWORKS`. But none that have a chain ID as the key.

Maybe we could do something like this instead:

```suggestion
  if (Object.values(BUILT_IN_INFURA_NETWORKS).map(({ chainId }) => chainId).includes(chainId)) {
```

Reply:
The same condition seems appropriate for the below change as well